solidity/test/bridge/Bridge.MovingFunds.test.ts

```diff
@@ -1929,42 +1996,1104 @@ describe("Bridge - Moving funds", () => {
+
+                                  // To run this scenario, we just mark the
+                                  // sweep request as processed using a stub
+                                  // method.
+                                  const beforeProofActions = async () => {
+                                    await bridge.setProcessedMovedFundsSweepRequest(
+                                      data.movedFundsSweepRequest
+                                        .walletPubKeyHash,
+                                      data.movedFundsSweepRequest
+                                    )
+                                  }
+
+                                  tx = runMovedFundsSweepScenario(
+                                    data,
+                                    beforeProofActions
+                                  )
+                                })
+
+                                after(async () => {
+                                  await restoreSnapshot()
+                                })
+
+                                it("should revert", async () => {
+                                  await expect(tx).to.be.revertedWith(
+                                    "Sweep request already processed"
+                                  )
+                                })
+                              }
+                            )
+
+                            context(
+                              "when the single input does refer to a known sweep request that belongs to another wallet",
+                              () => {
+                                const data: MovedFundsSweepTestData =
+                                  MovedFundsSweepWithoutMainUtxo
+
+                                before(async () => {
+                                  await createSnapshot()
+                                })
+
+                                after(async () => {
+                                  await restoreSnapshot()
+                                })
+
+                                it("should revert", async () => {
+                                  // To make this scenario happen, we just
+                                  // change the wallet in the test data' sweep
+                                  // request.
+                                  await expect(
+                                    runMovedFundsSweepScenario({
+                                      ...data,
+                                      movedFundsSweepRequest: {
+                                        ...data.movedFundsSweepRequest,
+                                        walletPubKeyHash:
+                                          "0x7ac2d9378a1c47e589dfb8095ca95ed2140d2726",
+                                      },
+                                    })
+                                  ).to.be.revertedWith(
+                                    "Sweep request belongs to another wallet"
```

Comment:
Practically, it should not happen given that the Bitcoin network should not allow for it, right?

Reply:
A sweeping wallet can unlock the received UTXO and lock it on another wallet's PKH. This is basically the resolution of the problem we detected in deposit sweeps: https://github.com/keep-network/tbtc-v2/issues/245

Reply:
Can we drop a comment about it in `MovingFunds.sol`, next to the validation:
```
require(
  sweepRequest.walletPubKeyHash == walletPubKeyHash,
  "Sweep request belongs to another wallet"
);
```
?

Reply:
https://github.com/keep-network/tbtc-v2/pull/244/commits/f6b962174ed59cc8b366222408af326ff09ebce7